pkg/pgmodel/migrations/sql/1_base_schema.up.sql

```diff
@@ -171,47 +182,91 @@ $func$
 LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_default_retention_period() TO prom_reader;
 
+--lock-order: metric table, data_table, series parent, series partition
+CREATE PROCEDURE SCHEMA_CATALOG.finalize_metric_creation()
+AS $proc$
+DECLARE
+    r RECORD;
+    created boolean;
+BEGIN
+    FOR r IN
+        SELECT *
+        FROM SCHEMA_CATALOG.metric
+        WHERE NOT creation_completed
```

Comment:
Should we order by random() to allow for better concurrency in the case where that happens?